server/sonar-server/src/main/java/org/sonar/server/platform/ws/MigrateDbSystemWsAction.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.platform.ws;
+
+import com.google.common.io.Resources;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.persistence.Database;
+import org.sonar.core.persistence.DatabaseVersion;
+import org.sonar.server.db.migrations.DatabaseMigration;
+
+import static org.sonar.server.db.migrations.DatabaseMigration.Status.NONE;
+import static org.sonar.server.db.migrations.DatabaseMigration.Status.SUCCEEDED;
+
+/**
+ * Implementation of the {@code migrate_db} action for the System WebService.
+ */
+public class MigrateDbSystemWsAction implements SystemWsAction {
+
+  private static final String UNSUPPORTED_DATABASE_MIGRATION_STATUS = "Unsupported DatabaseMigration status";
+
+  private final DatabaseVersion databaseVersion;
+  private final DatabaseMigration databaseMigration;
+  private final Database database;
+
+  public MigrateDbSystemWsAction(DatabaseVersion databaseVersion,
+    Database database,
+    DatabaseMigration databaseMigration) {
+    this.databaseVersion = databaseVersion;
+    this.database = database;
+    this.databaseMigration = databaseMigration;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    controller.createAction("migrate_db")
+      .setDescription("Migrate the database to match the current version of SonarQube." +
+        "<br/>" +
+        "Sending a POST request to this URL will start the DB migration. " +
+        "It is strongly advised to <strong>make a database backup</strong> before invoking this WS." +
+        "<br/>" +
+        "State values are:" +
+        "<ul>" +
+        "<li>NO_MIGRATION: DB is up to date with current version of SonarQube</li>" +
+        "<li>MIGRATION_NEEDED: DB needs to be upgraded in order for the current version of SonarQube to run</li>" +
```

Comment:
This state is misleading from a functional PoV. How is it possible as a migration was requested ? (I know the technical reason!)